src/intervals/interval_operations/pointwise_boolean.jl

```diff
@@ -0,0 +1,204 @@
+
+    This is different from what is described in sections 9.5 of the
+    IEEE Std 1788-2015 (Boolean functions of intervals), since we deal here
+    with what is needed for drop-in replacement of floats by intervals,
+    and not what is reasonnable to do with intervals in a separate execution
+    environment.
+
+    Essentially, julia is better at composability and we are more ambitious
+    that what the standard ever expected, so we have some extra problems
+    to solve.
+
+    We use a trait system, defining the operation with an extra
+    `PointwisePolitic` argument defining how it should be handled.
+
+    The default operations use `IntervalArithmetic.pointwise_politic()`,
+    so in order to change the default behavior this function has to be
+    redefined.
+
+    We define every operations using the default `PointwisePolitic{:ternary}()`
+    and then use it to define all the others.
+=#
+"""
+    PointwisePolitic{P}
+
+Define which politic we use to extend pointwise comparison of 
+
+Valid value for the politic identifier `P` are
+    - `:is_all` : A boolean operation is extended by asking "is it true for
+        all elements of the interval(s) involved".
+        This is self-consistent, but breaks the usual rules for negation.
+        For example with this politic, `((-1..3) == 0) == false` because it
+        answers the question "are all elements in (-1..3) equal to zero".
+        However its negation is `false` too: `((-1..3) != 0) == false`.
+        This *silently* breaks any code relying on such operation in conditional
+        statements like `if x == 0 ... else ... end`.
+    - `:interval` : A pointwise boolean operation `B` return the set of all
+        possible outcome as a `BooleanInterval`.
+        This is safe but very strict, always erroring when an interval is used
+        in a conditional statement.
+    - `:ternary` (default) : With this politics return `missing` when the
+        boolean operation does not return the same answer for all elements
+        of the involved interval(s).
+        This only causes error in conditional statements when hitting `missing`.
+        When it does not error it is safe.
+
+The current pointwise politic can be changed by overriding the function
+`IntervalArithmetic.pointwise_politic()`.
+
+Example
+=======
+
+          | (-1..3) > 2   | (-1..3) <= 2  | (-1..1) > 2 | (-1..1) < 2 |
+----------|---------------|---------------|-------------|-------------|
+:is_all   | false         | false         | false       | true        |
+:interval | [true, false] | [true, false] | [false]     | [true]      |
+:ternary  | missing       | missing       | false       | true        |
+
+"""
+struct PointwisePolitic{P} end
+
+pointwise_bool_operations = [
+    :(==), :(!=), :<, :(<=), :>, :(>=)
+]
+
+pointwise_bool_functions = [
+    :isinf, :isfinite, :isinteger, :iszero
+]
+
+## :ternary
+function ==(::PointwisePolitic{:ternary}, x::Interval, y::Interval)
```

Comment:
Docstrings are quite necessary for all `pointwise_bool_operations` and `pointwise_bool_functions`.